server/src/services/data_sources.service.ts

```diff
@@ -85,23 +85,25 @@ export class DataSourcesService {
     }, manager);
   }
 
-  async findOneByEnvironment(
-    dataSourceId: string,
-    organizationId: string,
-    environmentId?: string
-  ): Promise<DataSource> {
+  async findOneByEnvironment(dataSourceId: string, environmentId?: string): Promise<DataSource> {
     const dataSource = await this.dataSourcesRepository.findOneOrFail({
       where: { id: dataSourceId },
       relations: ['plugin', 'apps', 'dataSourceOptions'],
     });
 
     if (!environmentId && dataSource.dataSourceOptions?.length > 1) {
-      throw new NotAcceptableException('Environment id should not be empty');
+      //fix for env id issue when importing cloud/enterprise apps to CE
+      if (dataSource.dataSourceOptions?.length > 1) {
+        const env = await this.appEnvironmentService.get(dataSource.organizationId, null);
+        environmentId = env?.id;
+      } else {
+        throw new NotAcceptableException('Environment id should not be empty');
+      }
     }
 
     if (environmentId) {
       dataSource.options = (
-        await this.appEnvironmentService.getOptions(dataSourceId, organizationId, environmentId)
+        await this.appEnvironmentService.getOptions(dataSourceId, dataSource.organizationId, environmentId)
```

Comment:
organization id in data source is not mandatory field. will be present only for global data sources.